src/debug/debug.cc

```diff
@@ -4216,6 +4216,85 @@ std::string RecordReplayBasicValueContents(Handle<Object> value) {
   return "Unknown";
 }
 
+// Information in the dependency graph for a JS promise.
+struct PromiseDependencyGraphData {
+  // Graph node ID for the point the promise was created.
+  int new_node_id = 0;
+
+  // Graph node ID for the point the promise was settled.
+  int settled_node_id = 0;
+};
+
+typedef std::unordered_map<int32_t, PromiseDependencyGraphData> PromiseDependencyGraphDataMap;
+static PromiseDependencyGraphDataMap* gPromiseDependencyGraphDataMap;
+
+static PromiseDependencyGraphData&
+GetOrCreatePromiseDependencyGraphData(Isolate* isolate, Handle<Object> promise) {
+  v8::Isolate* v8_isolate = (v8::Isolate*) isolate;
+  int promise_object_id =
+    RecordReplayObjectId(v8_isolate, v8_isolate->GetCurrentContext(),
+                         v8::Utils::ToLocal(promise), /* allow_create */ true);
+
+  CHECK(IsMainThread());
```

Comment:
Nit: Move to top of function

Reply:
I left the check here because it ties in with the use of non-threadsafe global state immediately after this call.